x-pack/plugins/fleet/server/services/secrets.ts

```diff
@@ -405,9 +405,10 @@ export async function extractAndUpdateSecrets(opts: {
 
   const { toCreate, toDelete, noChange } = diffSecretPaths(oldSecretPaths, updatedSecretPaths);
 
+  const secretsToCreate = toCreate.filter((secretPath) => !!secretPath.value.value);
```

Comment:
`toCreate` is still being referenced at line [415](https://github.com/elastic/kibana/pull/172673/files#diff-da5efbb23b0aa4bda0850e130206b226217827821bbf0dcc0a6b343ab23618fbR415), being looped but using the filtered `createdSecrets`. This caused a bad access to `createdSecrets`

Stack trace
```
[2023-12-06T15:26:03.307+01:00][ERROR][plugins.fleet] TypeError: Cannot read properties of undefined (reading 'id')
    at forEach (secrets.ts:416:92)
    at Array.forEach (<anonymous>)
    at extractAndUpdateSecrets (secrets.ts:415:12)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at PackagePolicyClientImpl.update (package_policy.ts:754:28)
    at updatePackagePolicyHandler (handlers.ts:379:34)
    at core_versioned_route.ts:179:22
    at Router.handle (router.ts:228:30)
    at handler (router.ts:162:13)
    at exports.Manager.execute (/Users/romulo/Documents/repositories/security/elastic/kibana/node_modules/@hapi/hapi/lib/toolkit.js:60:28)
    at Object.internals.handler (/Users/romulo/Documents/repositories/security/elastic/kibana/node_modules/@hapi/hapi/lib/handler.js:46:20)
    at exports.execute (/Users/romulo/Documents/repositories/security/elastic/kibana/node_modules/@hapi/hapi/lib/handler.js:31:20)
    at Request._lifecycle (/Users/romulo/Documents/repositories/security/elastic/kibana/node_modules/@hapi/hapi/lib/request.js:371:32)
    at Request._execute (/Users/romulo/Documents/repositories/security/elastic/kibana/node_modules/@hapi/hapi/lib/request.js:281:9)
    ```

Reply:
Btw, it worked locally once I changed from 

```js
toCreate.forEach((secretPath, i) => {
```

to

```js
secretsToCreate.forEach((secretPath, i) => {
```

(for some reason I'm can't add the suggestion on this pr)

Reply:
you are right, I was not testing with fleet-server, so the secrets logic was not running. I'll fix in a new pr, merged this too early

Reply:
When we have a follow up fix for this can we backport the original PR + the follow up fix to 8.11 in case another patch is ever scheduled?

Reply:
https://github.com/elastic/kibana/pull/172695

Reply:
I can manually backport both PR's to 8.11 once the other is merged - I'll do it in a single manually created PR for simplicity.